crates/subspace-farmer/src/farmer_cache.rs

```diff
@@ -855,15 +847,17 @@ where
                         %piece_offset,
                         "Failed to write piece into cache"
                     );
+                    caches.push_stored_piece(old_key, offset);
```

Comment:
Why do we remember something as stored if it actually failed to store? Attempts to read it will result in errors.

It might be even worth doing some retries here in case other caches are still operational, but TODO is probably fine for now.